server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -266,13 +256,12 @@ public void rollGeneration() throws IOException {
         if (current.totalOperations() == 0 && primaryTermSupplier.getAsLong() == current.getPrimaryTerm()) {
             return;
         }
-        if (prepareForUpload(null)) {
-            performUpload(primaryTermSupplier.getAsLong(), null);
-        }
+        prepareAndUpload(primaryTermSupplier.getAsLong(), null);
     }
 
-    private boolean prepareForUpload(Long generation) throws IOException {
-        try (Releasable ignored = writeLock.acquire()) {
+    private boolean prepareAndUpload(Long primaryTerm, Long generation) throws IOException {
+        Releasable uploadReadLock = null;
```

Comment:
can we just use `readLock` instead of creating `uploadReadLock` ?

Reply:
Done